.github/workflows/backend.yml

```diff
@@ -0,0 +1,445 @@
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Avoid codecov error message related to SHA resolution:
+          # https://github.com/codecov/codecov-bash/blob/7100762afbc822b91806a6574658129fe0d23a7d/codecov#L891
+          fetch-depth: '2'
+
+      - name: Setup sentry env (python ${{ matrix.python-version }})
+        uses: ./.github/actions/setup-sentry
+        id: setup
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+          pip-cache-version: ${{ secrets.PIP_CACHE_VERSION }}
+          snuba: true
+          kafka: true
+
+      - name: Start symbolicator
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          echo $PWD
+          docker run \
+            -d \
+            -v $PWD/config/symbolicator/:/etc/symbolicator \
+            --network host \
+            --name symbolicator \
+            us.gcr.io/sentryio/symbolicator:nightly \
+            run -c /etc/symbolicator/config.yml
+          docker ps -a
+
+      - name: Run test
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make test-symbolicator
+
+      - name: Handle artifacts
+        uses: ./.github/actions/artifacts
+
+  typing:
+    needs: files-changed
+    name: backend typing
+    runs-on: ubuntu-20.04
+    timeout-minutes: 12
+    strategy:
+      matrix:
+        python-version: [3.8.12]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: ./.github/actions/setup-python
+        if: needs.changes.outputs.backend == 'true'
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Since we don't call the setup-sentry action we need to install libxmlsec1-dev
+      - name: Setup backend typing
+        if: needs.changes.outputs.backend == 'true'
+        env:
+          SENTRY_LIGHT_BUILD: 1
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends libxmlsec1-dev
+          python setup.py install_egg_info
+          pip install wheel # GitHub Actions does not have `wheel` installed by default
+          pip install -U -e ".[dev]"
+
+      - name: Run backend typing (${{ steps.setup.outputs.matrix-instance-number }} of ${{ strategy.job-total }})
+        if: needs.changes.outputs.backend == 'true'
+        run: |
+          make backend-typing
+
+  # This check runs once all dependant jobs have passed
+  # It symbolizes that all required Backend checks have succesfully passed
+  # This check is the only required Github check
+  backend-required-check:
+    needs:
+      [
+        api-docs,
+        backend,
+        cli,
+        lint,
+        migration,
+        plugins,
+        relay,
+        snuba,
+        symbolicator,
+        typing,
+      ]
+    if: ${{ needs.changes.outputs.backend == 'true' }}
```

Comment:
Eventually this condition will move to each one of the jobs and will be removed from here.